### PR TITLE
Ambient Light API

### LINKED ISSUE #26: Implement the Ambient Light API
Implement the API for the [Ambient Light](https://github.com/tock/tock/blob/master/capsules/extra/src/ambient_light.rs)

Documentation:
- libtock-rs overview - https://github.com/tock/libtock-rs/blob/master/doc/Overview.md
- Tock system calls - https://github.com/tock/tock/blob/master/doc/Syscalls.md
- Temperature API - https://github.com/tock/libtock-rs/tree/master/apis/temperature
- Buttons API - https://github.com/tock/libtock-rs/tree/master/apis/buttons
- microbit:v2 - https://microbit.org/new-microbit/

### PR DESCRIPTION
**Pull request by DanutAldea**
Fixes [#26](https://github.com/UPB-CS-OpenSourceUpstream/libtock-rs/issues/26)
Heavily inspired from the work done [here](https://github.com/UPB-CS-OpenSourceUpstream/libtock-rs/pull/5)
Added the api, an exemple, fake driver and tests.

### CODE REVIEW

File: examples/ambient_light.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+stack_size! {0x200}
+
+fn main() {
+    match AmbientLight::exists() {
+        Ok(()) => writeln!(Console::writer(), "ambient light driver available").unwrap(),
+        Err(_) => {
+            writeln!(Console::writer(), "ambient light driver unavailable").unwrap();
+            return;
+        }
+    }
```
Comment (alexandruradovici) [inline examples/ambient_light.rs:24]:
```suggestion
    if AmbientLight::exists().is_err() {
            writeln!(Console::writer(), "ambient light driver unavailable").unwrap();
            return;
    }
```

**Review by alexandruradovici — CHANGES_REQUESTED** [review]
Just simplify the example as suggested and this is good to go. Write a PR message.

**Comment by DanutAldea** [comment]
### Pull Request Overview
This PR adds an Ambient Light API, which includes:

- function for checking the existence of the driver
- functions for initating readings of the luminance, registering and unregistering a listener
- function for synchronous reading of values of luminance and 2 public functions for easier use of the API
- unit tests

Alongside the API, this PR adds a fake driver (and unit tests for it) for testing the API.

### Testing Strategy
This pull request was tested using unit tests made specifically for this API and fake driver.

### TODO or Help Wanted
This pull request still needs feedback / code review.
Help is needed to test this on a board, as I do not have access to one which has a light sensor module

**Review by alexandruradovici — DISMISSED** [review]
Looks really good, please send it to upstream and place a link to it. Hudson offered to test if it works on imix, you can tag him in the pull request message.

**Comment by DanutAldea** [comment]
https://github.com/tock/libtock-rs/pull/465

**Pull request merged**